cpp/src/join/semi_join.cu

```diff
@@ -34,10 +35,30 @@
 #include <thrust/copy.h>
 #include <thrust/distance.h>
 #include <thrust/sequence.h>
+#include <thrust/tuple.h>
+
+#include <cuco/static_map.cuh>
 
 namespace cudf {
 namespace detail {
 
+namespace {
+// Use the default cuco hasher.
+using Hash = cuco::detail::MurmurHash3_32<hash_value_type>;
+
+/**
+ * @brief Device functor to create a pair of hash value and index for a given row.
+ */
+struct new_pair_function {
+  __device__ __forceinline__ cudf::detail::pair_type operator()(size_type i) const noexcept
+  {
+    // Compute the hash value of row `i`
+    return cuco::make_pair<hash_value_type, size_type>(i, true);
```

Comment:
Is this a copy/paste remnant or something? There's not hash value being computed here.

Reply:
Yeah sorry there are a few bits of cruft lying around from early implementations when I copied more of the inner/left/full join logic than is really appropriate for semi/anti joins. Just cleaned that up.